java-frontend/src/main/java/org/sonar/java/bytecode/cfg/Instruction.java

```diff
@@ -81,6 +82,31 @@ public boolean hasReturnValue() {
     return Type.getMethodType(fieldOrMethod.desc).getReturnType() == Type.VOID_TYPE;
   }
 
+  @Override
+  public boolean equals(Object o) {
```

Comment:
This is a bit of "going the extra mile" but I think it's worth it : I would add a test for those implementation of equals/hashcode.